# World Happiness Report
"I've chosen to be happy because it's good for my health" - Voltaire

## Table of Contents 
### [1. Introduction](#introduction)
### [2. Data Sources](#data-sources)
### [3. Data Cleaning](#data-cleaning)
### [4. Data Transformation](#data-transformation)
### [Appendix](#chosethisappendix)



## Introduction

The main objective of this report is to understand how happiness relates to socio-economic factors, what are the countries that lead the rankings for happiness and what makes them stand out with respect to the rest. The first data set chosen for this report is the World Happiness Report, my choice to go for the most recent data set was dictated by the will to deepen the understanding, both mine and of the reader, of how happiness relates to subjective well-being and how social, urban and natural environments affect the happiness level. Data regarding all of these factors has been provided in data sets from year 2020 on. This data set alone can provide numerous insights on happiness, but is missing something which I consider crucial when studying human behaviour, which is the level of freedom each country provides, and this is where the second data set, The Human Freedom Index, comes into play. The Human Freedom Index analyzes an extraordinary number of factors and summarizes them into a score. A brief introduction to both data sets is provided below, while a detailed explanation of the key components of each data set is provided in the [appendix](#chosethisappendix-a-class"anchor"-id"1"a).



### Questions that drive the analysis

1. **Location Analysis**  

    * Where are the happiest countries located?

    * Do developed countries have an higher level of happiness when compared to less developed ones?

    * What region are the best and worst when comparing their human freedom scores?


2. **Component Analysis** 

    * How much does each factor impact on level of happiness?

    * What's the relationship between economic freedom, personal freedom and human freedom?

    * What country has the highest gap between personal freedom and personal freedom for women?

    * Is there any correlation between level of freedom and level of happiness?


3. **Time Analysis**

    * How has happiness level changed throughout the years?
    
    * Which country grew the most in terms of happiness level? Which one in terms of human freedom?

    * What's the world trend for happiness level, economic freedom, personal freedom, personal freedom for women and human freedom?

## Data Sources 


### About World Happiness Report

The World Happiness Report (WHR) is based on the science of wellbeing, which uses quantitative methods to understand how different life experiences influence people's happiness and quality of life. Their annual report is published by the United Nations Sustainable Development Solutions Network and is focused on monitoring wellbeing as well as its causes and consequences around the world. The report continues to gain global recognition as governments, organizations and civil society increasingly use happiness indicators to inform their policy-making decisions. Leading experts across fields - economics, psychology, survey analysis, national statistics, health, public policy and more - describe how measurements of well-being can be used effectively to assess the progress of nations. 

The happiness scores and rankings use data from the Gallup World Poll. The scores are based on answers to the main life evaluation question asked in the poll. This question, known as the Cantril ladder, asks respondents to think of a ladder with the best possible life for them being a 10 and the worst possible life being a 0 and to rate their own current lives on that scale. The scores are from nationally representative samples for the years 2020-2022 and use the Gallup weights to make the estimates representative.

>  Coverage : The World Happiness Report datasets cover a time span that goes from 2015 to 2021

### About The Human Freedom Index

The Human Freedom Index presents the state of human freedom in the world based on a broad measure that encompasses personal, civil, and economic freedom. Human freedom is a social concept that recognizes the dignity of individuals and is defined here as negative liberty or the absence of coercive constraint. Because freedom is inherently valuable and plays a role in human progress, it is worth measuring carefully. The Human Freedom Index is a resource that can help to more objectively observe relationships between freedom and other social and economic phenomena, as well as the ways in which the various dimensions of freedom interact with one another. The report is co​published by the Cato Institute and the Fraser Institute.

The HFI is the most comprehensive freedom index so far created for a globally meaningful set of countries and jurisdictions representing 98.1 percent of the world’s population. This seventh annual index uses 82 distinct indicators of personal and economic freedom in the following area : rule of law; security and safety; movement; religion; association, assembly and civil society; expression and information; relationships; size of government, legal system and property rights; sound money; freedom to trade internationally and regulation.

>  Coverage : The Human Freedom Index datasets cover a time span that goes from 2008 to 2019


## Data Cleaning
The first step is to import all the libraries we will need throughout the phases of data cleaning, data exploration, data transformation and data visualization, followed by the import of the datasets needed. One of the datasets, The Human Freedom Index one, already contains data for previous years, whereas data for the World Happiness Report are stored in different csv files based on the year of collection. 

In [1]:
import os
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

whr_location = "./Datasets/WHR/"
hfi_location = "./Datasets/HFI/HFI2021.csv"
whr_2015 = pd.read_csv(whr_location + "whr_2015.csv")
whr_2016 = pd.read_csv(whr_location + "whr_2016.csv")
whr_2017 = pd.read_csv(whr_location + "whr_2017.csv")
whr_2018 = pd.read_csv(whr_location + "whr_2018.csv")
whr_2019 = pd.read_csv(whr_location + "whr_2019.csv")
whr_2020 = pd.read_csv(whr_location + "whr_2020.csv")
whr_2021 = pd.read_csv(whr_location + "whr_2021.csv")
hfi = pd.read_csv(hfi_location)

In [2]:
whr_2021.head()

Country name Regional indicator  Ladder score  \
0      Finland     Western Europe         7.842   
1      Denmark     Western Europe         7.620   
2  Switzerland     Western Europe         7.571   
3      Iceland     Western Europe         7.554   
4  Netherlands     Western Europe         7.464   

   Standard error of ladder score  upperwhisker  lowerwhisker  \
0                           0.032         7.904         7.780   
1                           0.035         7.687         7.552   
2                           0.036         7.643         7.500   
3                           0.059         7.670         7.438   
4                           0.027         7.518         7.410   

   Logged GDP per capita  Social support  Healthy life expectancy  \
0                 10.775           0.954                     72.0   
1                 10.933           0.954                     72.7   
2                 11.117           0.942                     74.4   
3                 10.878           0.983                     73.0   
4                 10.932           0.942                     72.4   

   Freedom to make life choices  Generosity  Perceptions of corruption  \
0                         0.949      -0.098                      0.186   
1                         0.946       0.030                      0.179   
2                         0.919       0.025                      0.292   
3                         0.955       0.160                      0.673   
4                         0.913       0.175                      0.338   

   Ladder score in Dystopia  Explained by: Log GDP per capita  \
0                      2.43                             1.446   
1                      2.43                             1.502   
2                      2.43                             1.566   
3                      2.43                             1.482   
4                      2.43                             1.501   

   Explained by: Social support  Explained by: Healthy life expectancy  \
0                         1.106                                  0.741   
1                         1.108                                  0.763   
2                         1.079                                  0.816   
3                         1.172                                  0.772   
4                         1.079                                  0.753   

   Explained by: Freedom to make life choices  Explained by: Generosity  \
0                                       0.691                     0.124   
1                                       0.686                     0.208   
2                                       0.653                     0.204   
3                                       0.698                     0.293   
4                                       0.647                     0.302   

   Explained by: Perceptions of corruption  Dystopia + residual  
0                                    0.481                3.253  
1                                    0.485                2.868  
2                                    0.413                2.839  
3                                    0.170                2.967  
4                                    0.384                2.798

In [3]:
whr_2021.shape

(149, 20)

The **most recent dataset** has *149 rows* and *20 columns*

In [4]:
whr_2015.head()

Country          Region  Happiness Rank  Happiness Score  \
0  Switzerland  Western Europe               1            7.587   
1      Iceland  Western Europe               2            7.561   
2      Denmark  Western Europe               3            7.527   
3       Norway  Western Europe               4            7.522   
4       Canada   North America               5            7.427   

   Standard Error  Economy (GDP per Capita)   Family  \
0         0.03411                   1.39651  1.34951   
1         0.04884                   1.30232  1.40223   
2         0.03328                   1.32548  1.36058   
3         0.03880                   1.45900  1.33095   
4         0.03553                   1.32629  1.32261   

   Health (Life Expectancy)  Freedom  Trust (Government Corruption)  \
0                   0.94143  0.66557                        0.41978   
1                   0.94784  0.62877                        0.14145   
2                   0.87464  0.64938                        0.48357   
3                   0.88521  0.66973                        0.36503   
4                   0.90563  0.63297                        0.32957   

   Generosity  Dystopia Residual  
0     0.29678            2.51738  
1     0.43630            2.70201  
2     0.34139            2.49204  
3     0.34699            2.46531  
4     0.45811            2.45176

In [5]:
whr_2015.shape

(158, 12)

The **oldest dataset** has *158 rows* and *12 columns*.

In [6]:
names_dict = {}
for name in os.listdir(whr_location) : 
    names_dict[name] = pd.read_csv(whr_location + name)

for name in names_dict.keys():
    status_check = names_dict[name].isnull().sum().any()
    if status_check :
        print(f"Null values found in file {name}")
    else : 
        continue

Null values found in file whr_2018.csv


In [7]:
for column in whr_2018.columns:
    if whr_2018[column].isnull().sum().any():
        col = whr_2018.columns.get_loc(column)
        print(f"The error was found in column '{column}', which corresponds to column {col}")

The error was found in column 'Perceptions of corruption', which corresponds to column 8


In [8]:
row = int(np.where(whr_2018.iloc[:, col].isnull())[0])
print(whr_2018.iloc[row])

Overall rank                                      20
Country or region               United Arab Emirates
Score                                          6.774
GDP per capita                                 2.096
Social support                                 0.776
Healthy life expectancy                         0.67
Freedom to make life choices                   0.284
Generosity                                     0.186
Perceptions of corruption                        NaN
Name: 19, dtype: object


I found out that the missing data was only one entry for the United Arab Emirates, this is not likely to influence or skew in any way the analysis so I'll leave it as it is. 

In order to achieve a successful merge of the world happiness datasets, I want to check the countries which were in the old dataset and are not in the new one. First I make sure that there are no duplicates in the datasets : 

In [9]:
if (whr_2015["Country"].nunique() == whr_2015.shape[0]) and (whr_2020["Country name"].nunique() == whr_2020.shape[0]):
    print("No duplicates")

No duplicates


In [10]:
countries2015 = [name for name in whr_2015["Country"].unique()]
countries2021 = [name for name in whr_2021["Country name"].unique()]
missing_countries = [name for name in countries2015 if name not in countries2021]
print(missing_countries)

['Oman', 'Qatar', 'Taiwan', 'Suriname', 'Trinidad and Tobago', 'Hong Kong', 'Bhutan', 'Somaliland region', 'Macedonia', 'Sudan', 'Congo (Kinshasa)', 'Djibouti', 'Angola', 'Central African Republic', 'Syria']


Since there are differences in the number of country each datasets has, I'll make sure to keep only the ones that all the datasets have in common. I don't consider this a problem for the analysis because we still have data on more than 130 countries and those that will be left out are mainly small countries where data has been collected in 2015 but is not present in the 2021, probably due to difficulties in retrieving information.

In [11]:
hfi.head()

year  countries  ISO                         region  hf_score  hf_rank  \
0  2019    Albania  ALB                 Eastern Europe      8.14     43.0   
1  2019    Algeria  DZA     Middle East & North Africa      5.26    154.0   
2  2019     Angola  AGO             Sub-Saharan Africa      6.09    129.0   
3  2019  Argentina  ARG  Latin America & the Caribbean      7.38     74.0   
4  2019    Armenia  ARM        Caucasus & Central Asia      8.20     40.0   

   hf_quartile  pf_rol_procedural  pf_rol_civil  pf_rol_criminal  ...  \
0          2.0               5.97          4.76             4.26  ...   
1          4.0               5.21          5.64             4.35  ...   
2          4.0               2.72          4.43             3.60  ...   
3          2.0               6.83          5.94             4.35  ...   
4          1.0                NaN           NaN              NaN  ...   

   ef_regulation_business_adm  ef_regulation_business_bureaucracy  \
0                        5.65                                6.67   
1                        4.22                                2.22   
2                        2.94                                2.44   
3                        2.71                                5.78   
4                        5.17                                5.56   

   ef_regulation_business_start  ef_regulation_business_bribes  \
0                          9.74                           6.24   
1                          9.31                           2.58   
2                          8.73                           4.70   
3                          9.58                           6.53   
4                          9.86                           6.96   

   ef_regulation_business_licensing  ef_regulation_business_compliance  \
0                              5.62                               7.18   
1                              8.77                               7.03   
2                              7.92                               6.78   
3                              5.73                               6.51   
4                              9.30                               7.04   

   ef_regulation_business  ef_regulation  ef_score  ef_rank  
0                    6.85           7.70      7.81     31.0  
1                    5.69           5.84      4.90    162.0  
2                    5.59           5.97      5.50    153.0  
3                    6.14           5.99      5.50    153.0  
4                    7.32           7.82      8.03     15.0  

[5 rows x 125 columns]

In [12]:
hfi.shape

(1980, 125)

In [13]:
hfi.isnull().sum().sum()/np.product(hfi.shape) * 100

5.681212121212121

The Human Freedom Index has a lot of nulls, but also has numerous columns, which are very specific variables that act as components for the final scores in each category, (economical freedom, personal freedom etc.\*). As we can see approximately 6% of the data are missing, but once we drop the columns we're not interested in, the number may reduce drastically. 

\* More information on variables in the Human Freedom Index [here](#appendixappendix).

## Data Transformation

Problems to solve : 

1. The absence of a "year" column indicating when the data was collected 

2. During the years the naming convention for variables has changed

3. Both datasets have too many columns and they need to be dropped

4. Datasets for the WHR for years 2017, 2018 and 2019 contain only the variable "Country or region", the variable name is confusing since datasets for other years use region to indicate not the single country but bigger portions of land belonging to continents (e.g. Western Europe, North America, etc.)

5. Column are not aligned with column naming conventions, 

In [14]:
# Problem n.1

whr_2015["Year"] = 2015
whr_2016["Year"] = 2016
whr_2017["Year"] = 2017
whr_2018["Year"] = 2018
whr_2019["Year"] = 2019
whr_2020["Year"] = 2020
whr_2021["Year"] = 2021

In [15]:
print(f"""
      Columns 2017 : {list(whr_2017.columns)}\n
      Columns 2018 : {list(whr_2018.columns)}\n
      Columns 2019 : {list(whr_2019.columns)}\n
      Columns 2020 : {list(whr_2020.columns)}\n
      Columns 2021 : {list(whr_2021.columns)}
      """, end = " ")


      Columns 2017 : ['Country', 'Happiness.Rank', 'Happiness.Score', 'Whisker.high', 'Whisker.low', 'Economy..GDP.per.Capita.', 'Family', 'Health..Life.Expectancy.', 'Freedom', 'Generosity', 'Trust..Government.Corruption.', 'Dystopia.Residual', 'Year']

      Columns 2018 : ['Overall rank', 'Country or region', 'Score', 'GDP per capita', 'Social support', 'Healthy life expectancy', 'Freedom to make life choices', 'Generosity', 'Perceptions of corruption', 'Year']

      Columns 2019 : ['Overall rank', 'Country or region', 'Score', 'GDP per capita', 'Social support', 'Healthy life expectancy', 'Freedom to make life choices', 'Generosity', 'Perceptions of corruption', 'Year']

      Columns 2020 : ['Country name', 'Regional indicator', 'Ladder score', 'Standard error of ladder score', 'upperwhisker', 'lowerwhisker', 'Logged GDP per capita', 'Social support', 'Healthy life expectancy', 'Freedom to make life choices', 'Generosity', 'Perceptions of corruption', 'Ladder score in Dystopia',

In [16]:
# Problem n.2

new_2017 = whr_2017.rename(columns = {"Happiness.Score" : "Happiness Score"})
new_2018 = whr_2018.rename(columns = {"Country or region" : "Country", "Score" : "Happiness Score"})
new_2019 = whr_2019.rename(columns = {"Score" : "Happiness Score", "Country or region" : "Country"})
new_2020 = whr_2020.rename(columns = {"Country name" : "Country", "Regional indicator" : "Region", "Ladder score" : "Happiness Score"})
new_2021 = whr_2021.rename(columns = {"Country name" : "Country", "Regional indicator" : "Region", "Ladder score" : "Happiness Score"})

Below I'm using a function to understand what are the countries that every dataset has, getting rid of those which were recorded only for some years, this way it will be easier to merge them and data will be consistent throughout the years. I'm also dropping all the columns that aren't needed in the final dataset.

<a id='another_cell'></a>

In [17]:
# Problem n.3

countries_to_keep = pd.Series(list(set(whr_2015["Country"]) & set(whr_2016["Country"]) & set(new_2017["Country"]) \
    & set(new_2018["Country"]) & set(new_2019["Country"]) & set(new_2020["Country"]) & set(new_2021["Country"])))
columns_to_keep1 = ["Country", "Region", "Happiness Score", "Year"]
columns_to_keep2 = ["Country", "Happiness Score", "Year"]
new_2015 = whr_2015.loc[whr_2015["Country"].isin(countries_to_keep), columns_to_keep1]
new_2016 = whr_2016.loc[whr_2016["Country"].isin(countries_to_keep), columns_to_keep1]
new_2017 = new_2017.loc[new_2017["Country"].isin(countries_to_keep), columns_to_keep2]
new_2018 = new_2018.loc[new_2018["Country"].isin(countries_to_keep), columns_to_keep2]
new_2019 = new_2019.loc[new_2019["Country"].isin(countries_to_keep), columns_to_keep2]
new_2020 = new_2020.loc[new_2020["Country"].isin(countries_to_keep), columns_to_keep1]
new_2021 = new_2021.loc[new_2021["Country"].isin(countries_to_keep), columns_to_keep1]

Now I need to add the Regions to the datasets that only had the 'Country' variable, which are whr_2017, whr_2018 and whr_2019

In [18]:
# Problem n.4

new_2017["Region"] = new_2017.Country.map(new_2015.set_index("Country").Region)
new_2018["Region"] = new_2018.Country.map(new_2015.set_index("Country").Region)
new_2019["Region"] = new_2019.Country.map(new_2015.set_index("Country").Region)

final_whr = pd.concat([new_2015, new_2016, new_2017, new_2018, new_2019, new_2020, new_2021])
final_whr

Country              Region  Happiness Score  Year
0    Switzerland      Western Europe            7.587  2015
1        Iceland      Western Europe            7.561  2015
2        Denmark      Western Europe            7.527  2015
3         Norway      Western Europe            7.522  2015
4         Canada       North America            7.427  2015
..           ...                 ...              ...   ...
143       Malawi  Sub-Saharan Africa            3.600  2021
145     Botswana  Sub-Saharan Africa            3.467  2021
146       Rwanda  Sub-Saharan Africa            3.415  2021
147     Zimbabwe  Sub-Saharan Africa            3.145  2021
148  Afghanistan          South Asia            2.523  2021

[959 rows x 4 columns]

In [19]:
# Problem 

## CHOSETHISAppendix <a class="anchor" id="1"></a>

Appendix includes variable definitions for each dataset and some deep dives/extra visualization
-- Decide if you want the interactive dashboard to be on the table of contents or inserted at the end in the appendix

[Back to Table of Contents](#table-of-contents)

### World Happiness Report variable definition

* **Happiness score or subjective well-being** (variable name ladder) : is the national average response to the question of life evaluations. The English wording of the question is “Please imagine a ladder, with steps numbered from 0 at the bottom to 10 at the top. The top of the ladder represents the best possible life for you and the bottom of the ladder represents the worst possible life for you. On which step of the ladder would you say you personally feel you stand at this time?” This measure is also referred to as Cantril life ladder.

* **GDP per capita** variable name gdp 

* **Healthy Life Expectancy** (HLE) : healthy life expectancies at birth are based on the data extracted from the World Health Organization’s (WHO) Global Health Observatory data repository

* **Social support** (or having someone to count on in times of trouble) : is the national average of the binary responses (either 0 or 1) to the GWP question “If you were in trouble, do you have relatives or friends you can count on to help you whenever you need them, or not?”

* **Freedom to make life choices** : is the national average of responses to the GWP question “Are you satisfied or dissatisfied with your freedom to choose what you do with your life?”

* **Generosity** : is the residual of regressing national average of response to the GWP question “Have you donated money to a charity in the past month?” on GDP per capita.

* **Corruption Perception** : The measure is the national average of the survey responses to two questions in the GWP: “Is corruption widespread throughout the government or not” and “Is corruption widespread within businesses or not?” The overall perception is just the average of the two 0-or-1 responses. In case the perception of government corruption is missing, we use the perception of business corruption as the overall perception. The corruption perception at the national level is just the average response of the overall perception at the individual level.

* **Positive affect** : is defined as the average of three positive affect measures in GWP: laugh, enjoyment and doing interesting things in the Gallup World Poll waves 3-7. These measures are the responses to the following three questions, respectively: “Did you smile or laugh a lot yesterday?”, and “Did you experience the following feelings during A LOT OF THE DAY yesterday? How about Enjoyment?”, “Did you learn or do something interesting yesterday?”

* **Negative affect** : is defined as the average of three negative affect measures in GWP. They are worry, sadness and anger, respectively the responses to “Did you experience the following feelings during A LOT OF THE DAY yesterday? How about Worry?”, “Did you experience the following feelings during A LOT OF THE DAY yesterday? How about Sadness?”, and “Did you experience the following feelings during A LOT OF THE DAY yesterday? How about Anger?”

* **Gini of household income** : reported in the Gallup World Poll (variable name giniIncGallup). The income variable is described in Gallup’s “Worldwide Research Methodology And Codebook” as “Household Income International Dollars [...] To calculate income, respondents are asked to report their household income in local currency. Those respondents who have difficulty answering the question are presented a set of ranges in local currency and are asked which group they fall into. Income variables are created by converting local currency to International Dollars (ID) using purchasing power parity (PPP) ratios.” The gini measure is generated using STATA command ineqdec0 by WP5-year with sample weights.

* **Institutional trust**: The first principal component of the following five measures: confidence in the national government, confidence in the judicial system and courts, confidence in the honesty of elections, confidence in the local police force, and perceived corruption in business. This principal component is then used to create a binary measure of high institutional trust using the 75th percentile in the global distribution as the cutoff point; this way a country whose population tends to have a low level of institutional trust in the global distribution will have a low average institutional trust at the national level. This measure is not available for all countries since not all surveys in all countries ask all of the questions that are used to derive the principal component.

### The Human Freedom Index variable definition

* **Rule of law** :  The rule of law is an essential condition of freedom that protects the individual from coercion by others. Individual freedom is dependent on the rule of law, a broad concept that encompasses due process, equal treatment under the law, accountability of government officials and notions of fairness, predictability and justice. Indicators from the World Justice Project's (WJP) Rule of Law Index that are consistent with this definition of freedom are used. The indicators used not only rate the degree to which the rule of law may be undermined; they also measure negative rights. 

* **Security and safety** : The security and safety category measures actual crimes committed. It attempts to measure the degree to which people who have not violated the equal rights of others are physically assaulted, kidnapped or killed, or their physical integrity or safety is otherwise violated. Indicators include measurements for homicide rate; disappearances, conflicts and terrorism; violent conflicts and organized conflicts; level of fatalities and injuries that result from terrorism; freedom from torture and extent to which political killings are practiced and the degree to which they are incited and approved by the government.

* **Movement** : Freedom to travel. Indicators include freedom of foreign movement, freedom of movement for men and freedom of movement for women within the jurisdiction. 

* **Religion** : Free societies respect the right to practice a religion of one's choosing. Two components make up the measure of religious freedom, the first is based on two indicators that broadly rate the extent of freedom of religion in society, including the right to practice and choose one's religions. The second component gauges the repression of religious organizations by government.

* **Association, assembly, and civil society** : The freedom to associate and assemble with peaceful individuals or organizations of one's choice and to form or join organizations for political, commercial or other ends. This category is made up of the following components : the freedom of civil society organizations to enter and exit public life; freedom of assembly, including the ability to carry out peaceful protests; freedom to form and run political parties, including the extent to which parties are banned or face barriers and the degree of autonomy of opposition parties; and government repression of civil society.

* **Expression and information** : This category measures a broad range of freedoms, including that affecting personal expression, the press and use of the internet. Components in this category include press killed, press jailed, freedom of academic and cultural expression; harassment of journalists, government attempts to censor print and broadcast media and the internet, the extent of censorship among journalists and the press and the use of security or other laws to punish journalists, and other laws and regulations that influence media content. Indicators also capture the degree to which political pressure comes to bear on media coverage and takes into account state and non-state harassment of, and violence toward, journalists.

* **Relationships** : Freedom to have intimate and familial relationships with others and that those be based on equal rights between males and females. Components include a male-to-male relationship that gauges the extent to which sexual relationships between men are legal and a female-to-female indicator that gauges the same for relationships between women. Other components are divorce rates, inheritance rights and daughters in inheritance matters. The last component measures the prevalence of female genital mutilation among the population of women ages 15-49 years in a given country.

* **Size of government** : Measured as general government consumption spending as a percentage of total consumption. The rating for this component, as with many of the components, is designed to mirror the actual distribution of the raw data but on a 0-to-10 scale. The 1990 data were used to derive the maximum and minimum values for this component. Countries with a larger proportion of government expenditures received lower ratings. In contrast, as the ratio approaches the maximum value, the ratio moves toward 0. Components include government consumption, transfers and subsidies, government investment, top marginal tax rate and state ownerships of assets. 

* **Legal system and property rights** : Measures are based on factors such as judicial independence, impartial courts, protection of property rights, military interference in rule of law and politics, integrity of the legal system, legal enforcement of contracts, regulatory costs of the sale of real property and reliability of police. The rating for the Legal System and Property Rights category are adjusted to reflect inequalities in the legal treatment of women. 

* **Sound money** : This component measures the average annual growth of the money supply in the past five years minus average annual growth of real GDP in the past 10 years. Countries where growth of the money supply greatly exceeds growth of real output receive lower ratings. The M1 money supply (basically defined as checkable deposits plus currency in circulation) figures were used to measure the growth rate of the money supply. Components include money growth, standard deviation of inflation, inflation and freedom to own foreign currency bank accounts. 

* **Freedom to trade internationally** : Measures are based on tariffs (the amount of tax on international trade as a share of exports and imports), regulatory trade barriers, black-market exchange rates and controls of the movement of capital and people.

* **Regulation** : The rating for the Regulation category is calculated as the average of its three components—credit
market regulations, labor market regulations, and business regulations